src/clientsettingsdlg.cpp

```diff
@@ -1009,17 +1016,37 @@ void CClientSettingsDlg::OnEnableOPUS64StateChanged ( int value )
 
 void CClientSettingsDlg::OnFeedbackDetectionChanged ( int value ) { pSettings->bEnableFeedbackDetection = value == Qt::Checked; }
 
-void CClientSettingsDlg::OnCustomDirectoriesEditingFinished()
+void CClientSettingsDlg::OnDeleteCustomDirectoriesClicked()
 {
     if ( cbxCustomDirectories->currentText().isEmpty() && cbxCustomDirectories->currentData().isValid() )
```

Comment:
This needs to change to check whether the call was made for "delete current entry".

If it was, it then needs the existing check that current entry is valid and then it can delete the entry.  The old line 1019 should be all that's needed to do that.  It shouldn't need the `for ... while ...` loops.

It also doesn't need to return.  The `else` (i.e. "not the delete button") case will prevent anything unintended happening.